src/util/index.ts

```diff
@@ -1 +1,5 @@
 export const noop = (..._args: unknown[]): void => {};
+/**
+ * method returns an array of a given object's own enumerable property names, iterated in the same order that a normal loop would, with its types
+ */
+export const getKeys = Object.keys as <T extends object>(obj: T) => Array<keyof T>;
```

Comment:
uh.... what?? 😂 I might need a play by play of this in a pairing session.

Reply:
Ivan is a TypeScript king!

Reply:
by no means, but I can go over what this does, pretty cool in my opinion.

Reply:
i wrote a [ts-playground](https://www.typescriptlang.org/play?#code/MYewdgzgLgBA5gUygaQQTwjAvDA8gIwCsFgoA6Aa3UwENMAeAFRgQA8oEwATTEIkqAD4AFH0IAuGIwCU2QTACCAJyU009KmhAAzKYIDcAKChoADghgBhAJZLgAGwQAFJSFOYcAIlVdrAVwhPGAAfAALPYFtgPwBbbQQlTmAEILCIzg4lXG1tCCRPfRhDYzMLGztHAEkwbRBsGAAlEhAlLnpyh2dXdwAaGDBY-ASDYtBIWEiKhEkOqpq6nABvQxhVmB9-CEkARgAGHpW1yei4hKTpmH3D1eSwTOzcpB3DAF8jQzGIEEcyexA4YTHRwAbQA5BsAqCALrSd4EYikSjUQFRRzSMi1JQAURowAAFsJhBRZFh5Ms1jBPt8EL9-iipsCKDCjC9YcVhPCBEiMPTOrI6FZUV03BBgTCMS0cfjCcS5DByUdwF8fn8AUCEIzma82YZhIgUMj1dJ0ZipQSiSSyddKUrqbS1ULNWzWfogA) example yesterday to understand what was going on.